server/issue_test.go

```diff
@@ -279,6 +280,255 @@ func TestRouteShareIssuePublicly(t *testing.T) {
 	}
 }
 
+func TestShouldReceiveNotification(t *testing.T) {
+	s := ConnectionSettings{}
+	falseValue := false
+	tests := map[string]struct {
+		role         string
+		notification bool
+	}{
+		subCommandAssignee: {
+			role:         subCommandAssignee,
+			notification: falseValue,
+		},
+		subCommandMention: {
+			role:         subCommandMention,
+			notification: falseValue,
+		},
+		subCommandReporter: {
+			role:         subCommandReporter,
+			notification: falseValue,
+		},
+		subCommandWatching: {
+			role:         subCommandWatching,
+			notification: falseValue,
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			val := s.ShouldReceiveNotification(
+				tt.role,
+			)
+			assert.Equal(t, tt.notification, val)
+		})
+	}
+
+}
+
+func TestFetchConnectedUser(t *testing.T) {
+	tests := map[string]struct {
+		p           *Plugin
+		instanceID  types.ID
+		client      Client
+		connection  *Connection
+		wh          webhook
+		expectedErr error
+	}{
+		"No Field": {
+			p: &Plugin{
+				instanceStore: mockInstanceStore{},
+				userStore:     mockUserStore{},
+			},
+			instanceID: testInstance1.InstanceID,
+			client:     nil,
+			connection: nil,
+			wh: webhook{
+				JiraWebhook: &JiraWebhook{
+					Issue: jira.Issue{},
+				},
+			},
+			expectedErr: nil,
+		},
+		"Success": {
+			p: &Plugin{
+				instanceStore: mockInstanceStore{},
+				userStore:     mockUserStore{},
+			},
+			instanceID: testInstance1.InstanceID,
+			client:     testClient{},
+			connection: &Connection{
+				Settings: &ConnectionSettings{
+					Notifications:          true,
+					RolesForDMNotification: nil,
+				},
+			},
+			wh: webhook{
+				JiraWebhook: &JiraWebhook{
+					Issue: jira.Issue{
+						Fields: &jira.IssueFields{
+							Creator: &jira.User{},
+						},
+					},
+				},
+			},
+			expectedErr: nil,
+		},
+	}
+	for name, tt := range tests {
+		t.Run(name, func(t *testing.T) {
+			client, connection, error := tt.wh.fetchConnectedUser(
+				tt.p,
+				tt.instanceID,
+			)
+			assert.Equal(t, tt.connection, connection)
+			assert.Equal(t, tt.client, client)
+			if tt.expectedErr != nil {
+				assert.Error(t, tt.expectedErr, error)
+			}
+		})
+	}
+}
+
+func TestApplyReporterNotification(t *testing.T) {
+	api := &plugintest.API{}
+
+	api.On("LogError", mockAnythingOfTypeBatch("string", 13)...).Return(nil)
+
+	api.On("LogDebug", mockAnythingOfTypeBatch("string", 11)...).Return(nil)
+	p := Plugin{}
+	p.SetAPI(api)
+	p.instanceStore = p.getMockInstanceStoreKV(1)
+	p.userStore = getMockUserStoreKV()
+
+	tests := map[string]struct {
+		wh         *webhook
+		instanceID types.ID
+		reporter   *jira.User
+	}{
+		"No reporter": {
+			wh: &webhook{
```

Comment:
this is same for both tc and instant id so create a constants for this

Reply:
Now it is not same